src/libdistgen.cpp

```diff
@@ -118,35 +125,54 @@ double distgend_is_membound(distgend_configT config) {
 // INTERNAL FUNCTIONS
 //////////////////////////////////////////////////////////////////////////////////////////////////
 
-static double bench(distgend_configT config) {
-	double ret = 0.0;
-
-	omp_set_num_threads(system_config.number_of_threads);
+static void *thread_func(void *arg) {
+	thread_argsT *thread_args = (thread_argsT*)arg;
+	double *ret = (double*)calloc(1, sizeof(double));
 
-#pragma omp parallel reduction(+ : ret)
-	{
-		size_t tid = (size_t)omp_get_thread_num();
-		for (size_t i = 0; i < config.number_of_threads; ++i) {
-			if (tid == config.threads_to_use[i]) {
-				double tsum = 0.0;
-				u64 taCount = 0;
+	for (size_t i = 0; i < thread_args->config->number_of_threads; ++i) {
+		if (thread_args->tid == thread_args->config->threads_to_use[i]) {
+			double tsum = 0.0;
+			u64 taCount = 0;
 
-				const double t1 = wtime();
-				runBench(buffer[omp_get_thread_num()], iter, depChain, doWrite, &tsum, &taCount);
-				const double t2 = wtime();
+			const double t1 = wtime();
+			runBench(buffer[thread_args->tid], iter, depChain, doWrite, &tsum, &taCount);
+			const double t2 = wtime();
 
-				const double temp = taCount * 64.0 / 1024.0 / 1024.0 / 1024.0;
-				ret += temp / (t2 - t1);
-			}
+			const double temp = taCount * 64.0 / 1024.0 / 1024.0 / 1024.0;
+			*ret += temp / (t2 - t1);
 		}
 	}
 
+	pthread_exit((void*)ret);
+}
+
+
+static double bench(distgend_configT config) {
```

Comment:
> If possible I would try to avoid creating new threads during the benchmark run, as this increases the time we need to measure to get good results.

Is that really the case? Aren't we doing time measurements within the thread function?

>Have you checked if all core are used at the same time when `distgend_is_membound` is called.

What do you mean by *all* cores? The number of cores equal to `config.number_of_threads`? That ist the case, yes.